lib/common/xml.c

```diff
@@ -902,32 +902,47 @@ string2xml(const char *input)
     return xml;
 }
 
-xmlNode *
-stdin2xml(void)
+/*!
+ * \internal
+ * \brief Read from \c stdin until EOF or error
+ *
+ * \return Newly allocated string containing the bytes read from \c stdin, or
+ *         \c NULL on error
+ *
+ * \note The caller is responsible for freeing the return value using \c free().
+ */
+static char *
+read_stdin(void)
 {
-    size_t data_length = 0;
-    size_t read_chars = 0;
-
-    char *xml_buffer = NULL;
-    xmlNode *xml_obj = NULL;
+    char *buf = NULL;
+    size_t length = 0;
 
     do {
-        xml_buffer = pcmk__realloc(xml_buffer, data_length + PCMK__BUFFER_SIZE);
-        read_chars = fread(xml_buffer + data_length, 1, PCMK__BUFFER_SIZE,
-                           stdin);
-        data_length += read_chars;
-    } while (read_chars == PCMK__BUFFER_SIZE);
-
-    if (data_length == 0) {
-        crm_warn("No XML supplied on stdin");
-        free(xml_buffer);
-        return NULL;
+        size_t bytes_read = 0;
```

Comment:
don't actually need this var, was copying from `decompress_file()` -- will remove